react/react1/week3/to-do/src/components/FetchData.jsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import "react-datepicker/dist/react-datepicker.css";
+import Header from "./Header";
+import TodoItem from "./ToDoItem";
+import DatePicker from "react-datepicker";
+const FetchData = () => {
+  const [todos, setToDos] = useState([]);
+  const [edit, setEEdit] = useState(false);
+  const [newToDo, setNewToDo] = useState({});
+  const [deadLine, setDeadLine] = useState(new Date());
+  const NewToDo = (description) => {
+    setNewToDo({
+      id: Math.floor(Math.random() * (100 - 4 + 1) + 4),
+      description: description,
+      deadline: deadLine,
+      done: false,
+    });
+  };
+  const AddToDo = () => {
+    setToDos((previous) => [...previous, newToDo]);
+  };
+  const DeleteToDo = (id) => {
+    setToDos(todos.filter((eachToDo) => eachToDo.id !== id));
+  };
+  const editMode = (id) => {
+    setEEdit(!edit);
+    const index = todos.indexOf(todos.find((eachTodo) => eachTodo.id === id));
+    if (index !== -1) todos[index].isEdit = !todos[index].isEdit;
+    setToDos(todos);
+  };
+  const handleDesc = (desc, id) => {
+    todos.forEach((todo) =>
+      todo.id === id ? (todo.description = desc) : todo.description
+    );
+    const index = todos.indexOf(todos.find((eachTodo) => eachTodo.id === id));
+    if (index !== -1) todos[index].isEdit = !todos[index].isEdit;
+    setToDos(todos);
+    setEEdit(!edit);
+  };
```

Comment:
Why do you need the indexes to edit?

Reply:
I added an extra property in toDo list called isEdit 
and tried to update that boolean data if that particular toDo description is modified.this was not part of our homework ,I did it for self understanding to check how a key value pair in a object can be modified in react. 
I understand this is unwanted and buffer stuff .
Just for self understanding I tested this.

Thank you very much for your valuable time and feedback which will help mento learn further
🙏🏾.